macros/timeHistoryAnalyzer.py

```diff
@@ -0,0 +1,457 @@
+        """
+        if self.start == 0:
+            return 'never'
+        else:
+            return self._dates[self.start - 1]
+
+    def startString(self):
+        """Returns the date of the first scan in the range
+
+        Returns:
+            If the range includes the first available scan, returns 'first'.
+            Else returns a string containing the date of the scan, formatted as
+            %Y.%m.%d.%H.%M.
+        """
+        if self.start == 0:
+            return 'first'
+        else:
+            return self._dates[self.start]
+
+    def endString(self):
+        """Returns the date of the last scan in the range
+
+        Returns:
+            If the range includes the last available scan, returns 'never'. Else
+            returns a string containing the date of the scan, formatted as
+            %Y.%m.%d.%H.%M.
+        """
+        if self.end == len(self._dates):
+            return 'never'
+        else:
+            return self._dates[self.end - 1]
+
+    def afterEndString(self):
+        """Returns the date of the first scan after the range
+
+        Returns:
+            If the range includes the last available scan, returns 'none'. Else
+            returns a string containing the date of the scan, formatted as
+            %Y.%m.%d.%H.%M.
+        """
+        if self.end >= len(self._dates) - 1:
+            return 'none'
+        else:
+            return self._dates[self.end]
+
+    def scanCount(self):
+        """Returns the number of scans in the range"""
+        return self.end - self.start
+
+    def badMaskReasonScanCount(self):
+        """Returns the number of scans with non-zero maskReason in the range"""
+        return np.count_nonzero(self._maskReason[self.start:self.end])
+
+    def maskedScanCount(self):
+        """Returns the number of scans with mask set in the range"""
+        return np.count_nonzero(self._mask[self.start:self.end])
+
+    def maskedScanRatio(self):
+        """Returns the fraction of scans with mask set in the range"""
+        return float(self.maskedScanCount()) / self.scanCount()
+
+    def initialMaskReason(self):
+        """Returns the maskReason for the first scan in the range"""
+        return int(self._maskReason[self.start])
+
+    def allMaskReasons(self):
+        """Returns a maskReason bitmask that contains all the maskReasons in the
+        range"""
+        res = 0
+        for time in range(self.start, self.end):
+            res |= int(self._maskReason[time])
+        return res
+
+    def additionnalMaskReasons(self):
+        """Returns a maskReason bitmask that contains all the maskReaons in the
+        range but not in the first scan"""
+        return self.allMaskReasons() ^ self.initialMaskReason()
+
+    def noise(self):
+        """Returns a Numpy array containing the noise information for scans in
+        the range"""
+        return self._noise[self.start:self.end]
+
+def _findRangesMeta(data, vfat, channel, channelData, maxSkip):
```

Comment:
Why have a leading '_' here?  This is not a class method so having it follow syntax for private doesn't really make sense.

Reply:
It's a common convention in Python for module-private functions